frontend/src/metabase/static-viz/components/XYChart/shapes/BarSeries.tsx

```diff
@@ -66,6 +81,41 @@ export const BarSeries = ({
                 />
               );
             })}
+            {/* Render all data point values last so they stay on top of the chart elements making them more legible */}
+            {series.data.map((datum, index) => {
+              const groupX = xAccessor(datum);
+              const innerX = innerBarScale(seriesIndex) ?? 0;
+
+              const x = groupX + innerX;
+              const width = innerBarScale.bandwidth();
+
+              const yValue = yScale(getY(datum)) ?? 0;
+              const isNegative = getY(datum) < 0;
+
+              const bottomTextYPos =
```

Comment:
A calculation to determine whether the value would be flipped. We need to get the approx bottom Y position of the text and compare it with `xAxisYPos`

## Results
Look at the 4th bar
- ![Screen Shot 2022-08-23 at 6 57 16 PM](https://user-images.githubusercontent.com/1937582/186154135-0612afa4-895b-4de4-896d-42b62d59fea6.png)
- ![Screen Shot 2022-08-23 at 6 57 20 PM](https://user-images.githubusercontent.com/1937582/186154130-bbfbdb1c-0f7c-4ec5-b037-24db6a37a449.png)